mobile/src/app/pages/home/home.page.ts

```diff
@@ -46,17 +45,53 @@ export class HomePage implements OnInit {
   /** The `<map-rectangle>` component that displays on the map */
   @ViewChild(MapRectangle) rectangle: MapRectangle;
 
+  @ViewChild('placesRef') placesRef: GooglePlaceDirective;
+
   constructor(
     private http: HttpClient,
     private toastController: ToastController,
-    private storage: Storage
+    private storage: Storage,
+    private changeDetector: ChangeDetectorRef
   ) {}
 
-  async ionViewDidEnter() {
-    this.setting = await this.storage.get('setting:locationPreference');
+  ionViewDidEnter() {
+    google.maps.event.addListener(this.map.googleMap, 'tilesloaded', () => {
+      const input = document.getElementById(
+        'autocomplete-input'
+      ) as HTMLInputElement;
+
+      this.map.googleMap.controls[google.maps.ControlPosition.TOP_CENTER].push(
+        input
+      );
+
+      const searchBox = new google.maps.places.SearchBox(input);
+
+      /* Uses event listener instead of `onAddressChange` event so user doesn't
+        have to double-select their address. */
+      searchBox.addListener('places_changed', () => {
+        const places = searchBox.getPlaces();
+        const sizeFactor = 0.001; // TODO: Have `sizeFactor` change based on zoom.
+
+        const lat = places[0].geometry.location.lat();
+        const lng = places[0].geometry.location.lng();
+
+        this.map.googleMap.setCenter({ lat, lng });
+
+        if (this.showSelectionUI) {
+          this.bounds = {
+            north: places[0].geometry.location.lat() + sizeFactor,
+            south: places[0].geometry.location.lat() - sizeFactor,
+            east: places[0].geometry.location.lng() + sizeFactor,
+            west: places[0].geometry.location.lng() - sizeFactor
+          };
+
+          this.changeDetector.detectChanges();
```

Comment:
Check if. this is still necessary.

Reply:
Yes. if the user has a selection rectangle open then searches for an address, the rectangle will not appear until they navigate to the settings page and back.